### PR TITLE
use `oxnet::{ IpNet, Ipv4Net, Ipv6Net }`

### PR DESCRIPTION
**Pull request by ahl**

### CODE REVIEW

File: common/src/api/internal/shared.rs [hunk 1]
```diff
@@ -580,45 +581,43 @@ impl TryFrom<Vec<IpNet>> for IpAllowList {
...
-    use ipnetwork::{Ipv4Network, Ipv6Network};
+    use crate::api::internal::shared::AllowedSourceIps;
+    use oxnet::{IpNet, Ipv4Net, Ipv6Net};
     use std::net::{Ipv4Addr, Ipv6Addr};
 
     #[test]
     fn test_deserialize_allowed_source_ips() {
         let parsed: AllowedSourceIps = serde_json::from_str(
-            r#"{"allow":"list","ips":["127.0.0.1","10.0.0.0/24","fd00::1/64"]}"#,
+            r#"{"allow":"list","ips":["127.0.0.1/32","10.0.0.0/24","fd00::1/64"]}"#,
```
Comment (ahl) [inline common/src/api/internal/shared.rs:604]: @bnaecker this ok?

Comment (bnaecker) [reply common/src/api/internal/shared.rs:604]: Yep, that looks fine to me.

File: Cargo.toml [hunk 2]
```diff
@@ -331,6 +331,7 @@ omicron-certificates = { path = "certificates" }
 omicron-passwords = { path = "passwords" }
 omicron-workspace-hack = "0.1.0"
 oxlog = { path = "dev-tools/oxlog" }
+oxnet = { git = "https://github.com/oxidecomputer/oxnet", branch = "main" }
```
Comment (andrewjstone) [inline Cargo.toml:334]: Are we relying on the lockfile to keep this in sync? I usually don't like relying on a branch directly. Can we give it a 1.0 version tag or something?

File: clients/nexus-client/src/lib.rs [hunk 3]
```diff
@@ -419,33 +419,27 @@ impl TryFrom<types::ProducerEndpoint>
     }
 }
 
-impl TryFrom<&omicron_common::api::external::Ipv4Net> for types::Ipv4Net {
```
Comment (andrewjstone) [inline clients/nexus-client/src/lib.rs:422]: Are we planning to get rid of the generated `types` versions at some point?

File: common/src/address.rs [hunk 4]
```diff
@@ -72,6 +73,12 @@ pub const WICKETD_NEXUS_PROXY_PORT: u16 = 12229;
 
 pub const NTP_PORT: u16 = 123;
 
+/// The length for all VPC IPv6 prefixes
+pub const VPC_IPV6_PREFIX_LENGTH: u8 = 48;
+
+/// The prefix length for all VPC Sunets
```
Comment (andrewjstone) [inline common/src/address.rs:79]:
```suggestion
/// The prefix length for all VPC subnets
```

File: common/src/api/external/mod.rs [hunk 5]
```diff
@@ -1229,398 +1229,35 @@ impl DiskState {
...
-    fn from_str(s: &str) -> Result<Self, Self::Err> {
-        let net =
-            s.parse::<ipnetwork::IpNetwork>().map_err(|e| e.to_string())?;
-        match net {
-            ipnetwork::IpNetwork::V4(net) => Ok(IpNet::from(Ipv4Net(net))),
-            ipnetwork::IpNetwork::V6(net) => Ok(IpNet::from(Ipv6Net(net))),
-        }
+impl Ipv6NetExt for oxnet::Ipv6Net {
+    fn is_vpc_prefix(&self) -> bool {
+        // /// The prefix length for all VPC subnets
```
Comment (andrewjstone) [inline common/src/api/external/mod.rs:1252]: Should clean up this comment and dead code

File: nexus/db-queries/src/db/datastore/rack.rs [hunk 6]
```diff
@@ -1334,22 +1335,22 @@ mod test {
 
         let external_dns_ip = IpAddr::V4(Ipv4Addr::new(1, 2, 3, 4));
         let external_dns_pip = DNS_OPTE_IPV4_SUBNET
-            .nth(NUM_INITIAL_RESERVED_IP_ADDRESSES as u32 + 1)
```
Comment (andrewjstone) [inline nexus/db-queries/src/db/datastore/rack.rs:1337]: Very happy to get rid of the use of `as`.

**Review by andrewjstone — APPROVED** [review]
This looks great. Love the change from `network` to `prefix`. Just small nits from me.

File: Cargo.toml (hunk 2 above)
Comment (ahl) [reply Cargo.toml:334]: For now; I want to get to something resembling stable in oxnet, publish to crates.io and depend on that. We have several other branch = main deps

File: nexus/db-queries/src/db/datastore/rack.rs (hunk 6 above)
Comment (ahl) [reply nexus/db-queries/src/db/datastore/rack.rs:1337]: Good news: I added  more back in

File: clients/nexus-client/src/lib.rs (hunk 3 above)
Comment (andrewjstone) [reply clients/nexus-client/src/lib.rs:422]: Great!

File: Cargo.toml (hunk 2 above)
Comment (andrewjstone) [reply Cargo.toml:334]: Cool, works for me.

File: nexus/db-queries/src/db/datastore/rack.rs (hunk 6 above)
Comment (andrewjstone) [reply nexus/db-queries/src/db/datastore/rack.rs:1337]: lol

**Pull request merged**